include/wx/gtk/private/mediactrl.h

```diff
@@ -0,0 +1,46 @@
+#ifndef _WX_GTK_PRIVATE_MEDIACTRL_H_
+#define _WX_GTK_PRIVATE_MEDIACTRL_H_
+
+#ifdef GDK_WINDOWING_X11
+    #include <gdk/gdkx.h>
+#endif
+#ifdef GDK_WINDOWING_WAYLAND
+    #include <gdk/gdkwayland.h>
+#endif
+
+//-----------------------------------------------------------------------------
+// "wxGtkGetIdFromWidget" from widget
+//
+// Get the windows_id performing run-time checks If the window wasn't realized when Load was called, this is the
+// callback for when it is - the purpose of which is to tell
+// GStreamer to play the video in our control
+//-----------------------------------------------------------------------------
+extern "C" {
+inline gpointer wxGtkGetIdFromWidget(GtkWidget* widget)
+{
+    gdk_flush();
+
+    GdkWindow* window = gtk_widget_get_window(widget);
+    wxASSERT(window);
+
+#ifdef GDK_WINDOWING_X11
+#ifdef __WXGTK3__
+    if ( GDK_IS_X11_WINDOW(window) )
+#endif
+    {
+        return (gpointer)GDK_WINDOW_XID(window);
+    }
+#endif
+#ifdef GDK_WINDOWING_WAYLAND
+    if ( GDK_IS_WAYLAND_WINDOW(window) )
+    {
+        return (gpointer)gdk_wayland_window_get_wl_surface(window);
+    }
+#endif
+
+    wxSafeShowMessage(_("Fatal Error"), _("wxGtkGetIdFromWidget is only supported on Wayland and X11 currently.  You may be able to\nwork around this by setting environment variable GDK_BACKEND=x11 before\nstarting your program."));
```

Comment:
The only not completely trivial question I have is whether if we really need this `wxSafeShowMessage()`. It might not be a fatal error (ok, `wxMediaCtrl` is not going to work, but the rest of the program still might). And if we really need it, it would seem to be better to log this message from `wxGStreamerMediaBackend::CreateControl()` using the usual `wxLogError()`, IMO. E.g. do it if `SetupXOverlay()` returns `false`. But to be honest, I'd just not do it at all, I'm not sure the message really adds much.

Reply:
removed ;)